### PR TITLE
designs: async fifo test case

### PR DESCRIPTION
**Pull request by oharboe**
Asynchronous FIFOs appear in Chipyard MegaBoom. I thought would be useful to have a test case with reference to some decent documentation for this use-case.

### CODE REVIEW

**Comment by oharboe** [comment]
@maliberty I could use some help to articulate the .sdc files here...

**Comment by maliberty** [comment]
I think @tspyrou would be better able to help with complex SDC

**Comment by oharboe** [comment]
@tspyrou By design, the FIFO is designed such that the reader does not read from a location in the FIFO SRAM that is not stable. The FIFO SRAM is updated on the write clock, but the read of that FIFO location has a path from the read pointer which is updated on the read clock. That said, the needs to be a constraint between the writing and the reading.

The statement below, currently in use, is [discouraged](https://gist.github.com/brabect1/7695ead3d79be47576890bbcd61fe426#ignoring-cdc-paths) because it leaves the write/read relationship unconstrained:

```
set_clock_groups -asynchronous -group [get_clocks $clk1_name] -group [get_clocks $clk2_name]
```


image

**Comment by oharboe** [comment]
@tspyrou How can I refer to a register post-synthesis from the .sdc file? The name of a flip flop (register) is `_09645_` below. There are no other references to `_09645_` in the 1_synth.v file:

```
  DFFHQNx1_ASAP7_75t_R _09645_ (
    .CLK(wclk),
    .D(_01082_),
    .QN(_01034_)
  );
```

**Comment by oharboe** [comment]
@tspyrou I tried to use https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/353 to see if I could get some better names.

The situation is considerably improved:

```
  DFFHQNx1_ASAP7_75t_R \fifo_instance.fifomem.mem[10][18]$_DFFE_PP_  (
    .CLK(wclk),
    .D(_1548_),
    .QN(_0568_)
  );
```

**Review by tspyrou — APPROVED** [review]

**Comment by oharboe** [comment]
@maliberty Merge? This is a baseline for further development of the .sdc files once we have things in place w.r.t. signal names.

File: flow/designs/src/fifo/LICENSE [hunk 1]
```diff
@@ -0,0 +1,3 @@
+From http://www.sunburst-design.com/papers/CummingsSNUG2002SJ_FIFO1.pdf.
+
+No license or warranty is described in the paper.
```
Comment (maliberty) [inline flow/designs/src/fifo/LICENSE:3]: The paper itself will be covered by copyright.  Perhaps you can check with the author.

**Comment by oharboe** [comment]
@maliberty @tspyrou Force pushed to restart review process from scratch. If the computer says yes, then it is ready to go.

**Comment by maliberty** [comment]
Lots of
```
[WARNING STA-0353] set_output_delay not allowed on input port '...'.
```
and one
```
[WARNING STA-0354] set_input_delay relative to a clock defined on the same port/pin not allowed.
```

**Comment by oharboe** [comment]
@maliberty Deleted cruft, ready for review.

**Comment by oharboe** [comment]
force pushed because I accidentally included some changes to unrelated projects.

**Pull request merged**